Sources/Purchasing/Purchases/PurchasesType.swift

```diff
@@ -1008,6 +1009,35 @@ public protocol PurchasesSwiftType: AnyObject {
 
     #endif
 
+    /**
+     * Use this method only if you already have your own IAP implementation using StoreKit 2 and want to use
+     * RevenueCat's backend. If you are using StoreKit 1 for your implementation, you do not need this method.
+     *
+     * You only need to use this method with *new* purchases. Subscription updates are observed automatically.
+     *
+     * #### Example:
+     *
+     * ```swift
+     * let result = try await product.purchase(options: options)
+     * Purchases.shared.handleObserverModeTransaction(result)
+     * ```
+     *
+     * - Note: You need to finish the transaction yourself after calling this method.
+     *
+     * - Parameter purchaseResult: The ``StoreKit.Product.PurchaseResult`` of the product that was just purchased.
+     *
+     * - Throws: An error of type ``ErrorCode`` is thrown if a failure occurs while handling the purchase.
```

Comment:
This isn't guaranteed because Swift doesn't have typed throws yet.
See `PurchasesError.swift`, I've done a lot of work to make sure that we always ensure this.

You need to do this:
```swift
do {
    try purchasesOrchestratorMethod()
} catch {
  throw ErrorUtils.purchasesError(withUntypedError: self).asPublicError
}
```